tech-far-hub/gatsby-node.ts

```diff
@@ -65,22 +86,59 @@ export const createPages: GatsbyNode["createPages"] = async ({ graphql, actions,
     return;
   }
 
-  result.data.allMdx.edges.forEach(({ node }: { node: ITemplatedNode }) => {
+  const nodes: IEnhancedTemplatedNode[] = result.data.allMdx.edges.map(({ node }: { node: ITemplatedNode }) => {
     const parentDirs: string = node.parent.relativeDirectory;
-    const contentType: string = node.parent.relativeDirectory.split(path.sep)[0];
     const pageName: string = node.parent.name;
-    const pageIdentier: string = node.frontmatter.slug || pageName;
+    const pageIdentifier: string = node.frontmatter.slug || pageName;
+
+    const pagePath = pageName == "index" ? `/${parentDirs}/` : `/${parentDirs}/${pageIdentifier}/`;
+    const heading = node.frontmatter?.heading;
+    return {
+      ...node,
+      pagePath,
+      parentDirs,
+      pageName,
+      pageIdentifier,
+      heading,
+    };
+  });
+  const getHeadingForPath = (path: string): string => {
+    for (let node of nodes) {
+      if (path === "/") return "Home";
+      if (node.pagePath === path) return node.heading || path;
+    }
+    return path;
+  };
+
+  nodes.forEach((node: IEnhancedTemplatedNode) => {
+    const contentType: string = node.parent.relativeDirectory.split(path.sep)[0];
 
     // TODO: Make page-specific templates more general
-    const templateName = pageName === "index" ? "template-index.tsx" : "template-default.tsx";
-    const pagePath = pageName == "index" ? `${parentDirs}/` : `${parentDirs}/${pageIdentier}/`;
+    const templateName = node.pageName === "index" ? "template-index.tsx" : "template-default.tsx";
+    const pagePath = node.pagePath;
     const template = path.resolve("src", "pages", contentType, templateName);
+    const pathParts = node.pagePath.replace(/\/$/, "").split("/");
+    const breadCrumbPaths = pathParts.reduce(
+      (accumulator: string[], currentValue: string) => [
```

Comment:
While typing will catch most things, if unit testing is added, I might abstract this reducer's callback function into a separate function to allow that function to have its own test.

Reply:
That being said, nice use of chaining and keeping it functional!

Reply:
Good call --- I'll have to go back and add tests now that we're starting to solidify what we're building.